File: packages/python/src/opensms/resources/pricing.py
"""The ``pricing`` resource."""

from __future__ import annotations

from typing import Optional

from ..models import PriceList
from ._base import Resource


class Pricing(Resource):
    """Workspace price list. Accessed as ``client.pricing``."""

    def get(self, *, product: Optional[str] = None, country: Optional[str] = None) -> PriceList:
        """Prices for ``product`` (``sms``, ``lookup`` or ``number_monthly``),
        optionally for one ``country``."""
        return self._t.request("GET", "/v1/pricing", query={"product": product, "country": country})

File: packages/ruby/lib/opensms/resources/pricing.rb
# frozen_string_literal: true

require_relative "base"

module Opensms
  module Resources
    # The +pricing+ resource. Accessed as +client.pricing+.
    class Pricing < Base
      # GET /v1/pricing -> PriceList.
      # @param params [Hash] :product (sms|lookup|number_monthly), :country (ISO2)
      def get(params = nil, **kwargs)
        http_get("/v1/pricing", Models.build(merge(params, kwargs), %i[product country]))
      end
    end
  end
end
